package.json

```diff
@@ -65,7 +65,7 @@
     "qr-image": "^3.2.0",
     "string-natural-compare": "^3.0.1",
     "uuid": "^9.0.0",
-    "vue": "~2.6.14",
+    "vue": "~3.3.7",
```

Comment:
i need  "vue": "~2.7.15", not 3.3 here

Reply:
double check if 2.7 works. we had to downgrade in the past because it did not work with the virtual scroller component used for the contacts list

Reply:
We use `vue-virtual-scroller` in Talk with Vue 2.7, it should work.

We also tried to use `vue-virtual-scroll-list` and it worked in 2.7 if I remember correctly.

If it doesn't work, a minor upgrade may help, I guess.

Reply:
It doesn't work for this app https://github.com/nextcloud/contacts/pull/3117

Reply:
> If it doesn't work, a minor upgrade may help, I guess.

Both packages are at their latest version already

Reply:
> TypeError: this.$route is undefined

is the specific error we get in `ContactsListItem.vue` wrapped with `vue-virtual-scroll-list`.

Reply:
If I refactor the code to props taking the $route info from the parent component, I get a `Unknown custom element: <router-link> - did you register the component correctly? For recursive components, make sure to provide the "name" option.`. So my suspicion is that the virtual scroll list doesn't register the vue router on its items.

Reply:
Maybe that is actually a problem of `NcListItem` which is used in the scroller and can be a `router-link` but does not register it.

Reply:
Fair point. In my understanding all components are known to the Vue instance. If we construct the instance with the router, router-link should be a known component everywhere. :thinking: https://v3.router.vuejs.org/api/#component-injected-properties

Reply:
For the record.

The problem was that in the app Webpack resolved imported Vue as `vue.esm.js` while from `vue-virtual-scroll-list` as `vue.common.js`.

So there were 2 different copies of Vue in the bundle, and Vue-Router was installed only in one copy.

Dirty fix: 
```js
webpackConfig.resolve.alias = {
	vue$: 'vue/dist/vue.esm.js',
}
```